spec/system/require_edit_reason_spec.rb

```diff
@@ -46,18 +50,21 @@ def open_composer
 
     it "warns when no reason added" do
       find(".save-or-cancel .create.disabled").click
-      
       expect(page).to have_css(".edit-enforcer.--highlight")
       expect(page).to have_css(".d-icon-triangle-exclamation")
       expect(page).to have_css(".save-or-cancel .create.disabled")
     end
   end
 
   context "grace period handling" do
+    before { unset_force_edit_grace_period }
+
     it "skips enforcement within grace period" do
       open_composer
+      expect(page).to have_css("#reply-control.open")
```

Comment:
Part 3: it's sneaky, but `have_no_css` will pass if the spec is ran too quickly, i.e. before the component has had a chance to apply on the composer. This here makes sure the composer is fully open before running the checks below it.